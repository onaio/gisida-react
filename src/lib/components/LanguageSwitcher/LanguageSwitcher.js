import React from 'react';
import { connect } from 'react-redux';
import { Actions } from 'gisida';
import './LanguageSwitcher.scss';
const mapStateToProps = (state, ownProps) => {
  const { APP, LOC } = state;
  return {
    APP,
    LOC,
    ...ownProps,
  };
};

class LanguageSwitcher extends React.Component {
  constructor(props) {
    super(props);
  }

  onLocationClick(e) {
    const { dispatch } = this.props;
    e.preventDefault();
    if (e && e.currentTarget && e.currentTarget.dataset && e.currentTarget.dataset.id) {
      dispatch(Actions.setLocation(this.props.mapId, e.currentTarget.dataset.id));
    }
  }

  render() {
    const { APP } = this.props;
    return APP["translation"].length ? 
    (<a href="#" class="dropdown">
        <span class="glyphicon glyphicon-globe"></span>
        <div class="dropdown-content">
            { APP["translation"].map(lingua =>
                <a href={`${window.location.href.split('/')[2]}/${lingua.param}`}>{lingua.label}</a>
            )}
        </div>
      </a>) : null;
  }
}

export default connect(mapStateToProps)(LanguageSwitcher);
