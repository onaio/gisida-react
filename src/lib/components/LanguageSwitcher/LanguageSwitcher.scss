.dropbtn {
  background-color: #4caf50;
  color: white;
  padding: 16px;
  font-size: 16px;
  border: none;
}

.language-dropdown {
  font-size: 18px;
  text-align: center;
  width: 34px;
  background: #fff;
  z-index: 500;
  right: 8px;
  top: 328px;
  display: block;
  position: absolute;
  line-height: 1.8;
  color: #222;
  border: 2px solid #cdcdcd;
  border-radius: 7px;
}

.language-dropdown-content {
  display: none;
  position: absolute;
  background-color: #f1f1f1;
  min-width: 110px;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
  z-index: 1;
  right: 0;
  top: 34px;
}

.language-dropdown-content span {
  color: black;
  padding: 12px 16px;
  text-decoration: none;
  display: block;
  font-size: 10px;
}

.language-dropdown-content span:hover {
  background-color: #ddd;
  text-decoration: none;
}

.language-dropdown:hover .language-dropdown-content {
  display: block;
}

.language-dropdown:hover .dropbtn {
  background-color: #3e8e41;
}
.summary-language-switcher {
  display: flex;
  justify-content: flex-end;
  padding: 6px;
}
.summary-language-switcher select {
  border-radius: 2px;
  margin-top: 10px;
  text-indent: 6px;
  -o-text-overflow: '';
  text-overflow: '';
  width: 9%;
  height: 30px;
  font-size: 15px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent
    url(http://cdn1.iconfinder.com/data/icons/cc_mono_icon_set/blacks/16x16/br_down.png) no-repeat
    135px center;
}

/* Select Wrapper (for befores & afters) */

$select-color: #b6b6b6;
.custom-select {
  /* wrapper style */
  position: relative;
  display: block;
  max-width: 400px;
  min-width: 180px;
  margin: 5px auto;

  select {
    /* Select Reset */
    border: 1px solid $select-color;
    outline: none;
    background: transparent;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    border-radius: 0;
    margin: 0;
    display: block;
    width: 100%;
    cursor: pointer;

    /* custom select appearance */
    padding: 12px 55px 15px 15px;
    font-size: 14px;
  }
  /* the down arrow */
  &:after {
    position: absolute;
    right: 0;
    top: 0;
    width: 50px;
    height: 100%;
    line-height: 50px;
    content: '\25BC';
    text-align: center;
    z-index: -1;
    background: $select-color;
    color: #fff;
    font-size: 18px;
  }
}
