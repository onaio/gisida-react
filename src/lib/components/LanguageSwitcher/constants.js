export const LANGUAGESWITCHERCONFIGS = 'languageSwitcherConfigs';
export const SELECT_LANGUAGE = "Select Language";