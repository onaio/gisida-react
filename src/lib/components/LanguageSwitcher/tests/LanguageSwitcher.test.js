import React from 'react';
import LanguageSwitcher from '../LanguageSwitcher';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

describe('LanguageSwitcher', () => {
  const initialState = {
    APP: {
      translation: [
        { label: 'English', param: 'en' },
        { label: 'ជនជាតិខ្មែរ', param: 'khmer' },
      ],
      languageSwitcherConfigs: {
        English: {
          param: 'en',
        },
        ជនជាតិខ្មែរ: {
          param: 'khmer',
        },
      },
      languageTranslations: {
        khmer: {
          'Indicator 2.2 RCFs ': 'ជនជាតិខ្មែរ ជនជាតិខ្មែរ',
        },
      },
      mapLanguageTranslation: false,
    },
  };
  const mockStore = configureStore();

  it('renders without crashing', () => {
    const store = mockStore(initialState);
    shallow(
      <Provider store={store}>
        <LanguageSwitcher />
      </Provider>
    );
  });

  it('component renders correctly', () => {
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <LanguageSwitcher />
      </Provider>
    );
    expect(wrapper.text()).toMatchSnapshot('Text snapshot');
    expect(wrapper.html()).toMatchSnapshot('Elements snapshot');
  });
});
