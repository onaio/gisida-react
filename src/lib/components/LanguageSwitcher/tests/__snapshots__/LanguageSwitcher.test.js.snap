// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcher component renders correctly: Elements snapshot 1`] = `"<div class=\\"custom-select\\"><select><option value=\\"select language\\">Select Language</option><option value=\\"English\\">English</option><option value=\\"ជនជាតិខ្មែរ\\">ជនជាតិខ្មែរ</option></select></div>"`;

exports[`LanguageSwitcher component renders correctly: Text snapshot 1`] = `"Select LanguageEnglishជនជាតិខ្មែរ"`;
